guides/provider/README.md

```diff
@@ -1,330 +1,458 @@
-host: http://provider.$PROVIDER_DOMAIN
-attributes:
-  - key: region
-    value: us-west-demo-$(whoami)
-EOF
-```
+The `--private-key` option lets Ansible know what SSH private key to authenticate with. If you're using a different private key be sure and update this parameter to point at the location of the private key used to access all hosts in the cluster.
 
-### Fund your provider account
+The playbook `cluster.yml` contains all the necessary steps to build the Kubernetes cluster. When the cluster is complete, on the master node there should be a file stored at `/root/.kube/config` which is used to connect to and authenticate with the Kubernetes. Treat this file as a secret password which you do not share. A copy of this file must be used to configure the Akash provider. 
 
-View your address with
 
-```sh
-akash keys show provider -a --home=$AKASH_HOME --keyring-backend=$AKASH_KEYRING_BACKEND 
-```
+### Known issues
 
-You can fund the address at the testnet [faucet](https://akash.vitwit.com/faucet).
+When the above example is followed, Kubespray uses Calico for the networking of Kubernetes.It is an extremely common error to have Kubespray fail on a step where a script named `calioctl.sh` is being run. This appears to be a bug in Kubespray. The only solution known at this time is to wait around 3 minutes then rerun Kubespray. On the second run, Kubespray complete successfully. 
 
-Ensure you have funds with:
+### Dense Provider considerations
 
-```sh
-akash query bank balances --home=$AKASH_HOME --node=$AKASH_NODE "$(akash keys show $AKASH_PROVIDER_KEY -a --home=$AKASH_HOME --keyring-backend=$AKASH_KEYRING_BACKEND )"
-```
+Kubernetes uses a networking model that assigns a single IP address to each deployed pod. Each node in a Kubernetes cluster has a range of IP addresses available to it. This means that there is a limit to the number of pods that can be hosted on a single node.
 
-### Create Akash Provider
+For providers with small nodes, this pods per node limit is not a concern. For providers running nodes with dense compute configurations (for example, multisocket CPU servers) this can become a limiting factor to utilization of the node.
 
-Register your provider on the Akash Network:
+To increase the pods per node limit, 3 steps need to be taken:
 
-*Replace `--from` flag value with your own*
+1. Increase `kube_network_node_prefix` to assign additional IP addresses to a single node.
+2. Increase `kube_pods_subnet` to allow more pods in the kubernetes cluster
+3. Increase `kube_service_addresses` since more services are expected to be running.
 
-```sh
-akash tx provider create provider.yaml --from $AKASH_PROVIDER_KEY --home=$AKASH_HOME --keyring-backend=$AKASH_KEYRING_BACKEND --node=$AKASH_NODE --chain-id=$AKASH_CHAIN_ID
-```
+In this example, these values can be configured in `kubespray/inventory/defi/group_vars/k8s_cluster/k8s-cluster.yml`
 
-### Configure provider services client
+<!-- TODO - add notes about adding CRD -->
```

Comment:
This is explained below when `kubectl` is used to invoke `crd.yaml` , so we can drop the TODO